meta.go

```diff
@@ -0,0 +1,351 @@
+package process
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/derision-test/glock"
+)
+
+// Meta is a wrapper around a process or initializer value. This wrapper ensures that
+// the configured receiver's methods are only called once and not called from an invalid
+// state (e.g. Start called before Init or after a failed Init).
+type Meta struct {
+	wrapped     interface{}
+	options     *metaOptions
+	logger      Logger
+	mu          sync.Mutex
+	initialized bool
+	running     bool
+	stopping    bool
+	stopped     chan struct{}
+}
+
+var defaultClock = glock.NewRealClock()
+
+func newMeta(wrapped interface{}, configs ...MetaConfigFunc) *Meta {
+	options := &metaOptions{
+		health:        NewHealth(),
+		contextFilter: func(ctx context.Context) context.Context { return ctx },
+		logger:        NilLogger,
+		initClock:     defaultClock,
+		startupClock:  defaultClock,
+		stopClock:     defaultClock,
+		shutdownClock: defaultClock,
+		finalizeClock: defaultClock,
+	}
+
+	for _, f := range configs {
+		f(options)
+	}
+
+	return &Meta{
+		wrapped: wrapped,
+		options: options,
+		logger:  options.logger.WithFields(options.metadata),
+		stopped: make(chan struct{}),
+	}
+}
+
+// Wrapped returns the underlying receiver.
+func (m *Meta) Wrapped() interface{} {
+	return m.wrapped
+}
+
+// Name returns the initializer or process's configured name or `<unnamed>` if one was not
+// supplied.
+func (m *Meta) Name() string {
+	if m.options.name == "" {
+		return "<unnamed>"
```

Comment:
Could this be "<unnamed: SomeReflectedName>" to make it easier to find which components aren't named?

Reply:
Updated.